website/pages/docs/developers/creating-new-plugin/javascript-source.mdx

```diff
@@ -0,0 +1,172 @@
+
+Resolvers are functions associated with a table that get called when it's time to populate data for that table. There are two types of resolvers:
+
+#### Table resolvers
+
+Table resolvers are responsible for fetching data from the third-party API. In JavaScript, a table resolver in an object that matches the [`TableResolver`](https://github.com/cloudquery/plugin-sdk-javascript/blob/7a4c4a54d7fcfb266d7cd75c371b86a378ac7969/src/schema/table.ts#L16) TypeScript type. For top-level tables, `resolve` will only be called once per multiplexer client. For dependent tables, the resolver will be called once for each parent row, and the parent resource will be passed in as well. (More on this, and multiplexers, shortly.)
+
+#### Column resolvers
+
+[Column resolvers](https://github.com/cloudquery/plugin-sdk-javascript/blob/7a4c4a54d7fcfb266d7cd75c371b86a378ac7969/src/schema/column.ts#L13) are responsible for mapping data from the third-party API into the columns of the table. In most cases, you will not need to implement this, as the SDK will automatically map data from the struct passed in by the table resolver to the columns of the table. But in some cases, you may need to implement a custom column resolver to fetch additional data or do custom transformations.
+
+### Multiplexers
+
+[Multiplexers](https://github.com/cloudquery/plugin-sdk-javascript/blob/7a4c4a54d7fcfb266d7cd75c371b86a378ac7969/src/schema/table.ts#L18) are a way to parallelize the fetching of data from the third-party API. Some top-level tables require multiple calls to fetch all their data. For example, a sync for the GitHub source plugin that fetches data for multiple organizations, will need to make one call per organization to list all repositories. By multiplexing over organizations, these top-level queries can also be done in parallel. Each table defines the multiplexer that it should use. The CloudQuery plugin SDK will then call the table resolver once for each client in the multiplexer. Many plugins will not need to use multiplexers, but they are useful for plugins that need to fetch data for multiple accounts, organizations, or other entities.
+
+## Apache Arrow JavaScript SDK
+
+Like other CloudQuery SDK the JavaScript SDK uses [Apache Arrow] as part of the underlying CloudQuery type system.
+When writing a source plugin you should not install the Apache JavaScript SDK via `npm` and instead use the version of the SDK that is bundled with the CloudQuery JavaScript SDK for maximum compatibility.
+See [example in the Airtable plugin](https://github.com/cloudquery/cloudquery/blob/cb4a46ed9bdd205f19f3d92636a40914aedc6b24/plugins/source/airtable/src/tables.ts#L12).
+
+## Creating Your First Plugin
+
+As the JavaScript SDK is still new, we don't yet have a plugin scaffold generator. For now, we recommend starting by copying the code from an [existing JavaScript plugin](https://github.com/cloudquery/cloudquery/tree/cb4a46ed9bdd205f19f3d92636a40914aedc6b24/plugins/source/airtable).
```

Comment:
In this specific case, let's link this to the `main` branch so whatever users end up copying has all the latest fixes?

Reply:
Done